jrnl/controller.py

```diff
@@ -218,45 +305,16 @@ def search_mode(args: "Namespace", journal: Journal, **kwargs) -> None:
         _display_search_results(**kwargs)
 
 
-def _write_in_editor(config: dict, template: str | None = None) -> str:
+def _write_in_editor(config: dict, prepopulated_text: str | None = None) -> str:
```

Comment:
Nice change with `prepopulated_text` change instead of `template`. It's good to root out metonymies like this from the code.

Reply:
> I feel particularly hesitant about the environment variable change. Could you remove that part from this PR? I think we need a broader conversation about how jrnl could handle environment variables, and I think that will take some time, but I don't want it to hold up this nice `--template` change.

Sure. The purpose of the env var was to provide some shorthand for creating a new jrnl from a template (instead of having to `cd` to the template directory and then run `$ jrnl --template LOCAL_TEMPLATE`). Since `jrnl` already supports `XDG`, does it make sense to create `$XDG_CONFIG_HOME/jrnl/templates` if it does not exist, and use that as the shorthand templates directory instead?

> I think we could use some BDD tests for `--template` as well. There's a new `template.feature` file in there that could use these tests. Let me know if you'd like any help with that.

I agree that tests are worthwhile here. I saw the BDD I wanted to get your eyes on the PR before I invested time in writing tests.

Reply:
> Since `jrnl` already supports `XDG`, does it make sense to create `$XDG_CONFIG_HOME/jrnl/templates` if it does not exist, and use that as the shorthand templates directory instead?

Sorry for the delay in response here. I think your suggestion is a great compromise and I'm all for it. Though I think we should use `$XDG_DATA_HOME/jrnl/templates` instead since that fits the XDG spec and we already use `$XDG_DATA_HOME/jrnl` as the default path for the default journal.